### ISSUE TITLE
Net-SNMP V5.9.x: No increment of snmpInTotalSetVars

### ISSUE THREAD

**Issue by kschuster267** [net-snmp/net-snmp#429]
Hello,

I have an issue with the OID "snmpInTotalSetVars" in the latest version of Net-SNMP (V5.9.x). Setting a writeable OID will not lead to increment this counter. 
![0297655e-4e91-4203-8da6-8e4a70d34a5a](https://user-images.githubusercontent.com/94831237/173811319-294bbb29-5811-4f43-99fe-f36a100ed24c.png)

When I was using Net-SNMP V5.7.3 it was working. What did happen between these 2 version? I did not change any config.
![5b7d5bea-90b8-4136-8805-e2e6ad7ce613](https://user-images.githubusercontent.com/94831237/173811458-0f1c50ab-db13-41e7-a761-891f818fe885.png)

Would be great that somebody is enlightening me :)
Thanks!

**Comment by mindcrunch4u** [comment]
are you sure the "set" was successful?

in: `net-snmp-5.9.1/agent/snmp_agent.c`
at: `netsnmp_wrap_up_request()`
```
        if (status == SNMP_ERR_NOERROR)
            snmp_increment_statistic_by(
#ifndef NETSNMP_NO_WRITE_SUPPORT
                (asp->pdu->command == SNMP_MSG_SET ?
                 STAT_SNMPINTOTALSETVARS : STAT_SNMPINTOTALREQVARS),
#else
                STAT_SNMPINTOTALREQVARS,
#endif
                count_varbinds(asp->pdu->variables));
```
indicates that **only a successful request triggers the counter to increment.**

while `snmpInSetRequests` doesn't require a successful "set" to be incremented.
in: `net-snmp-5.9.1/agent/snmp_agent.c`
at: `handle_set()`
```
    if (!(asp->pdu->flags & UCD_MSG_FLAG_ONE_PASS_ONLY)) {
        switch (asp->mode) {
        case MODE_SET_BEGIN:
            snmp_increment_statistic(STAT_SNMPINSETREQUESTS);
            asp->rw = WRITE;    /* WWW: still needed? */
            asp->mode = MODE_SET_RESERVE1;
            asp->status = SNMP_ERR_NOERROR;
            break;
```

**Comment by kschuster267** [comment]
Yes, the writeable value was written correctly, thats why I am confused why this parameter is not incrementing like in the version before.
